### ISSUE TITLE
Add component backtrace on invalid argument exception

### ISSUE THREAD

**Issue by sascha-egerer** [sitegeist/fluid-components#139]
If an invalid argument is passed to a component an exception is thrown:

`The argument "image" defined in "Foo\Bar\FluidComponents\Component\Image" has declared type "SMS\FluidComponents\Domain\Model\Image", but "NULL" was give`

Sometimes it's really hard to figure out where the error occurs. It would be very helpful to have a component stack backtrace in the exception.
`The argument "image" defined in "Foo\Bar\FluidComponents\Component\Image" has declared type "SMS\FluidComponents\Domain\Model\Image", but "NULL" was give in Foo\Bar\FluidComponents\Module\Layout\Page\Default -> Foo\Bar\FluidComponents\Module\Footer -> Foo\Bar\FluidComponents\Module\LogoBox -> Foo\Bar\FluidComponents\Component\Link -> Foo\Bar\FluidComponents\Component\Image`

**Comment by s2b** [comment]
Good idea! This could be based on #144 which introduces a data structure for component metadata.

**Comment by sascha-egerer** [comment]
Yeah I habe a patch ready based on a previous patch of named slots. I can push it one rebased on master
